lib/Bindings/Python/circt/dialects/_hw_ops_ext.py

```diff
@@ -71,28 +71,18 @@ def __getattr__(self, name):
     # Check for the attribute in the arg name set.
     if name in self.operand_indices:
       index = self.operand_indices[name]
-      return self.instance.inputs[index]
+      value = self.instance.inputs[index]
+      return BuilderValue(value, self, index)
```

Comment:
Would it be possible to avoid the use of `BuilderValue` and instead return the `OpOperand`? I'm thinking then `connect` could work with ops built the 'normal' way and re-assign their operands. I _think_ that would make this generic in one fell swoop. Is it possible to get the `OpOperand` through `OpView`?

Reply:
Just looked and it doesn't look like you can get an actual `OpOperand` since `PyOpOperandList::getElement()` returns `PyValue`.

Reply:
yeah... there is no OpOperand in Python, and it seemed a bit much to add since we already have the `operation.operands[...] = ...` API

Reply:
yeah. It'd be nice to be able to do `connect(op.inputA, value)` on arbitrary ops rather than `connect(op, "inputA", value)`.

Reply:
> I'm thinking beyond instance builder.

My current thinking is there should be 1) a builder for ops which use ODS to name operands, which supports access through `__getattr__`, and 2) a generic builder that works by index for the variadic things in the `comb` dialect. So the current two argument `connect` would work the same for ops with ODS operand names (reg, unary and binary ops, etc), and a three argument `connect(op, index, value)` would be necessary to support the variadic ops.

Reply:
Actually, the index is carried in the destination argument, so the two argument `connect` should be sufficient, as long as the new builders return proper BuilderValues.

Reply:
I'm still pondering several possibilities here but I think it'll be more productive to see what you come up and can make work.